src/main/java/com/thealgorithms/sorts/QuickSort.java

```diff
@@ -3,75 +3,77 @@
 import static com.thealgorithms.sorts.SortUtils.*;
 
 /**
- * @author Varun Upadhyay (https://github.com/varunu28)
- * @author Podshivalov Nikita (https://github.com/nikitap492)
+ * Implementation of the QuickSort algorithm.
+ * 
+ * @author Manish Raj (https://github.com/manishraj27)
  * @see SortAlgorithm
  */
 class QuickSort implements SortAlgorithm {
 
     /**
-     * This method implements the Generic Quick Sort
+     * Sorts the input array using QuickSort algorithm.
      *
-     * @param array The array to be sorted Sorts the array in increasing order
+     * @param array The array to be sorted. Sorts the array in increasing order.
+     * @return The sorted array.
      */
     @Override
     public <T extends Comparable<T>> T[] sort(T[] array) {
-        doSort(array, 0, array.length - 1);
+        quickSort(array, 0, array.length - 1);
```

Comment:
__Error Handling__: You should add a check to handle cases where the array is null or has only one element, returning the array directly in such cases.
```suggestion
        if (array == null || array.length <= 1) {
            return array;
        }
        quickSort(array, 0, array.length - 1);
```